content/tutorials/build-apps/build-a-desktop-wallet-in-javascript.md

```diff
@@ -0,0 +1,615 @@
+
+Here we define the libraries our application will use in the `dependencies` section as well as shortcuts for running our application in the `scripts` section. To install those dependencies, run the following command:
+```console
+npm install
+```
+
+This installs the Electron Framework, the xrpl.js client library and a couple of helpers we are going to need for our
+application to work. 
+
+### 1. Hello XRP Ledger
+
+**Full code for this step:** 
+[`1_hello.js`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/desktop-js/1_hello.js),
+[`view/1_preload.js`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/desktop-js/view/1_preload.js),
+[`view/1_hello.html`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/desktop-js/view/1_hello.html),
+[`view/1_renderer.js`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/desktop-js/view/1_renderer.js).
+
+Our first step is to write a "Hello World" like application, that interacts on a very basic level with the XRP Ledger 
+and displays some high-level information about the current ledger state on the screen. Nothing too fancy so far, as we 
+will take care of styling and GUI related coding in a later step:
+
+![Screenshot: Step 1, hello world equivalent](img/javascript-wallet-1.png)
+
+First, we will create an entrypoint for our application, for this we create the file `1_hello.js` with the following content:
+
+`1_hello.js`
+{{ include_code("_code-samples/build-a-wallet/desktop-js/1_hello.js", language="js") }}
+
+This code has two parts: one that creates the application window and one that calls the XRP Ledger API's [ledger method](ledger.html). The code then broadcasts an event with the API response as the payload. The frontend picks up this event and uses the payload to display the index of most recently validated ledger.
+
+To display our results to the user, we need to create the view components that we specified in the `createWindow()` function. For this, we will create a `view` folder and add the following files there:
+
+<!-- MULTICODE_BLOCK_START -->
+
+*view/1_preload.js*
+{{ include_code("_code-samples/build-a-wallet/desktop-js/view/1_preload.js", language="js") }}
+
+*view/1_hello.html*
+{{ include_code("_code-samples/build-a-wallet/desktop-js/view/1_hello.html", language="html") }}
+
+*view/1_renderer.js*
+{{ include_code("_code-samples/build-a-wallet/desktop-js/view/1_renderer.js", language="js") }}
+
+<!-- MULTICODE_BLOCK_END-->
+
+The file `view/1_preload.js` does the main wiring of the application. The file `view/1_hello.html` defines the template part of the view. The file `view/1_renderer.js` contains scripts used in the template; you could also have put these in the HTML file, but it's a good practice to separate them so the code is more readable.
```

Comment:
These explanations seem like they'd make sense to include as one-sentence summaries of the files in-line with the code to copy.

Reply:
Removed